pandas/core/internals/blocks.py

```diff
@@ -2002,6 +2002,16 @@ class ObjectValuesExtensionBlock(ExtensionBlock):
     def external_values(self):
         return self.values.astype(object)
 
+    def _can_hold_element(self, element: Any) -> bool:
+        if is_valid_nat_for_dtype(element, self.dtype):
+            return True
+        if isinstance(element, list) and len(element) == 0:
+            return True
```

Comment:
why this? i dont think we have this in other _can_hold_element methods

Reply:
Without it, `pandas/tests/extension/test_interval.py::TestSetitem::test_setitem_empty_indexer` will not pass. It is also used on L2223. If there's a better way to handle this please let me know, but I did borrow this from 2223